types/lhr/lhr.d.ts

```diff
@@ -138,6 +140,37 @@ declare module Result {
     descriptions: Record<string, string>;
   }
 
+  /**
+   * Entity classification for the run, for resolving URLs/items to entities in report.
+   * The two lookup tables (LUT) below provide space-optimized, O(1) index lookup into entities.
+   */
+  interface EntityClassification {
+    /** All entities (1st and 3rd party) discovered during the run */
+    entities: Array<Entity>;
+    /** Name of the first-party entity */
+    firstParty?: string;
+    /** Entity-name to entity index lookup table  */
+    nameLUT: Record<string, number>;
+    /** URL origin to entity index lookup table */
+    originLUT: Record<string, number>;
+  }
+
+  /**
+   * An entity that's either recognized by third-party-web or made up by Lighthouse.
+   */
+  interface Entity {
+    /** Name of the entity. Maps to third-party-web unique name for recognized entities and a root domain name for the unrecognized. */
+    name: string;
+    /** Homepage URL for a recognized entity, if available in third-party-web. */
+    homepage?: string;
+    /** Category name that the entity belongs to, if available. */
+    category?: string;
+    /** Is this entity the first party? */
+    isFirstParty?: boolean;
+    /** Is this entity recognized by third-party-web? */
+    isUnrecognized?: boolean;
+  }
```

Comment:
- of these properties, only `name` seems to be used anywhere. But I can see homepage/category/isUnrecognized being useful just for querying, and this object is very small (rounds to 0% of total LHR json) so it's fine to have.
- `isFirstParty` is only true once, and seems to be always the first item (is that right?); and is just the entity that has the same name as `firstParty` string in parent object. If this is for simpler HTTP archive queries, that's fine–is that the intended purpose?

Reply:
That is true re: usage of fields, at least in this PR. The subsequent PR that introduces grouping, will use all the other fields introduced here. I could move it as part of that PR, but it wouldn't matter much (unless we ship only this change with 10.0).

`entityClassification.firstParty` provides the name/unique identifier of the first party to perform a direct lookup. The `boolean` value `entity.isFirstParty` will be present only on the first-party entity object (will be omitted for all else).  This is likely the least amount and most flexible storage optimized for lookup as well. 

Both the fields cater to report-generation and possibly to JSON/HA consumers. The string is used by `report-ui-features`'s third-party filter at the moment.